index.bs

```diff
@@ -188,6 +286,37 @@ Every [=Document=] has an associated [=ordered set=] of <dfn>previously reported
     When asked to [=mark paint timing=] given a [=Document=] |document| as input, perform the following steps:
     1. If the [=document=]'s [=Document/browsing context=] is not [=paint-timing eligible=], return.
     1. Let |paintTimestamp| be the [=current high resolution time=].
+    1. Let |paintedImages| be a new [=ordered set=]
+    1. Let |paintedTextNodes| be a new [=ordered set=]
+    1. For each |imagePendingRenderingTriple| in |doc|'s [=images pending rendering=] list:
+        1. Let |imageRequest| be the [=image request=] of |imagePendingRenderingTriple|.
+        1. If |imageRequest| is fully decoded, then run the following steps:
+            1. Append |imagePendingRenderingTriple| to |paintedImages|.
+            1. Remove |imagePendingRenderingTriple| from |doc|'s <a>images pending rendering</a> list.
+    1. For each {{Element}} |element| in |doc|'s <a>descendants</a>:
+        1. If |element| is contained in |doc|'s <a>set of elements with rendered text</a>, continue.
+        1. If |element|'s <a>set of owned text nodes</a> is empty, continue.
+        1. <a for="set">Append</a> |element| to |doc|'s <a>set of elements with rendered text</a>.
+        1. [=set/Append=] |element| to |paintedTextNodes|.
+    1. Run [=the paint timing steps=] given |document|, |paintTimestamp|,
```

Comment:
We don't usually have this pluggable pattern in specs... it's better to call the specific ones

Reply:
Is there a better spec pattern that can be used to provide a hook here for related specs?
LCP and ET both append steps to the paint timing steps in the current model, for instance. I'd like to have a way to make this extensible, without having to reference those specs here.

Reply:
Referencing them here would be the pattern. That's how we do that in the HTML spec for instance, it references many other specs that might be at a "higher level".

Reply:
Yea I would simply call:
- Report paint timing given ...
- Report element timing given...